instrumentation/src/worker.ts

```diff
@@ -0,0 +1,99 @@
+  const workerLogger = logger.child({ label: 'worker' });
+  const workflowLogger = logger.child({ label: 'workflow' });
+  const activityLogger = logger.child({ label: 'activity' });
+
+  // @@@SNIPSTART typescript-core-telemetry-options
+  // Configure Rust Core to export SDK logs, metrics and (optionally) internal traces
+  await Core.install({
+    // Install a logger to collect logs generated by Node.js Workers and Rust Core.
+    logger: new DefaultLogger('DEBUG', (entry) => {
+      workerLogger.log({
+        level: entry.level.toLowerCase(),
+        message: entry.message,
+        timestamp: Number(entry.timestampNanos / 1_000_000n),
+        ...entry.meta,
+      });
+    }),
+    // Telemetry options control how logs, metrics and traces are exported out of Rust Core
+    telemetryOptions: {
+      // To export metrics and traces using the OpenTelemetry Collector, use `oTelCollectorUrl`.
+      // see https://opentelemetry.io/docs/collector/getting-started/ for more information.
+      //
+      // To expose a port for Prometheus to collect metrics from Core, use `prometheusMetricsBindAddress`
+      // You can verify metrics are exported with `curl -fail localhost:9464/metrics`
+      prometheusMetricsBindAddress: '0.0.0.0:9464',
+      // A string in the env filter format specified here:
+      // https://docs.rs/tracing-subscriber/0.2.20/tracing_subscriber/struct.EnvFilter.html
+      //
+      // Which determines what tracing data is collected in the Core SDK
+      tracingFilter: 'temporal_sdk_core=DEBUG',
+      // What level, if any, logs should be forwarded from Rust Core to the Node.js logger
+      logForwardingLevel: 'DEBUG',
+    },
+  });
+  // @@@SNIPEND
+
+  // The Worker side of our logger sinks, forwards logs from Workflows to a Winston logger
+  const sinks: InjectedSinks<LoggerSinks> = {
+    logger: {
+      debug: {
+        fn(workflowInfo, message, meta) {
+          workflowLogger.child(workflowInfo).debug(message, meta);
+        },
+      },
+      info: {
+        fn(workflowInfo, message, meta) {
+          workflowLogger.child(workflowInfo).info(message, meta);
+        },
+      },
+      error: {
+        fn(workflowInfo, message, meta) {
+          workflowLogger.child(workflowInfo).error(message, meta);
+        },
+      },
+      warn: {
+        fn(workflowInfo, message, meta) {
+          workflowLogger.child(workflowInfo).warn(message, meta);
+        },
+      },
+    },
+  };
+
+  // @@@SNIPSTART typescript-worker-full-logging-setup
+  // Create a worker that will use the Core instance installed above
+  const worker = await Worker.create({
+    workflowsPath: require.resolve('./workflows'),
+    activities,
+    taskQueue: 'instrumentation',
+    // Install interceptors
+    interceptors: {
+      activityInbound: [(ctx) => new ActivityInboundLogInterceptor(ctx, activityLogger)],
+      workflowModules: [require.resolve('./workflows/interceptors')],
```

Comment:
Should we support custom bundling?

Reply:
Maybe we should but it'd make the sample a bit less concise.
We should perhaps consider having all-in-one dev + prod ready sample.